base_location/views/partner_view.xml

```diff
@@ -1,24 +1,49 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <odoo>
+    <record id="view_address_partner_form" model="ir.ui.view">
+        <field name="name">res.partner.address.form.cb</field>
+        <field name="model">res.partner</field>
+        <field name="arch" type="xml">
+            <form>
+                <div class="o_address_format">
+                    <field name="street" placeholder="Street..."
+                           class="o_address_street"/>
+                    <field name="street2" placeholder="Street 2..."
+                           class="o_address_street"/>
+                    <field name="city" placeholder="City"
+                           class="o_address_city"/>
+                    <field name="state_id" class="o_address_state"
+                           placeholder="State"
+                           options="{&quot;no_open&quot;: True}"/>
+                    <field name="zip" placeholder="ZIP" class="o_address_zip"/>
+                    <field name="country_id" placeholder="Country"
+                           class="o_address_country"
+                           options="{&quot;no_open&quot;: True, &quot;no_create&quot;: True}"/>
+                </div>
+            </form>
+        </field>
+    </record>
 
     <record id="view_partner_form" model="ir.ui.view">
         <field name="name">res.partner.zip_id.2</field>
         <field name="model">res.partner</field>
-        <field name="inherit_id" ref="base.view_partner_form" />
+        <field name="inherit_id" ref="base.view_partner_form"/>
         <field name="arch" type="xml">
             <field name="city" position="before">
                 <field name="zip_id"
```

Comment:
The field should not be editable for contact persons, as it happens will other fields in this circumstance.
![image](https://user-images.githubusercontent.com/7683926/31438797-ef62e9d4-ae89-11e7-9cc5-cccdbef9135f.png)


Just add `attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"`